File: subprojects/frida-node/examples/crash_reporting.ts
import frida from "frida";

async function main() {
    const device = await frida.getUsbDevice();
    device.processCrashed.connect(onProcessCrashed);

    const session = await device.attach("Hello");
    session.detached.connect(onSessionDetached);

    console.log("[*] Ready");
}

function onProcessCrashed(crash: frida.Crash) {
    console.log("[*] onProcessCrashed() crash:", crash);
    console.log(crash.report);
}

function onSessionDetached(reason: frida.SessionDetachReason, crash: frida.Crash | null) {
    console.log("[*] onDetached() reason:", reason, "crash:", crash);
}

main()
    .catch(e => {
        console.error(e);
    });

File: subprojects/frida-python/examples/crash_reporting.py
import sys

import frida


def on_process_crashed(crash):
    print("on_process_crashed")
    print("\tcrash:", crash)


def on_detached(reason, crash):
    print("on_detached()")
    print("\treason:", reason)
    print("\tcrash:", crash)


device = frida.get_usb_device()
device.on("process-crashed", on_process_crashed)
session = device.attach("Hello")
session.on("detached", on_detached)
print("[*] Ready")
sys.stdin.read()
